airbyte-integrations/bases/source-acceptance-test/unit_tests/test_backward_compatibility.py

```diff
@@ -0,0 +1,887 @@
+                    },
+                }
+            ),
+            pytest.raises(AssertionError),
+            id="Changing a field type from a list to an empty list should fail.",
+        ),
+        pytest.param(
+            ConnectorSpecification(
+                connectionSpecification={
+                    "type": "object",
+                    "properties": {
+                        "my_int": {"type": ["str"]},
+                    },
+                }
+            ),
+            ConnectorSpecification(
+                connectionSpecification={
+                    "type": "object",
+                    "properties": {
+                        "my_int": {"type": "int"},
+                    },
+                }
+            ),
+            pytest.raises(AssertionError),
+            id="Changing a field type should fail from a list to string with different value should fail.",
+        ),
+        pytest.param(
+            ConnectorSpecification(
+                connectionSpecification={
+                    "type": "object",
+                    "properties": {
+                        "my_int": {"type": ["str"]},
+                    },
+                }
+            ),
+            ConnectorSpecification(
+                connectionSpecification={
+                    "type": "object",
+                    "properties": {
+                        "my_int": {"type": "str"},
+                    },
+                }
+            ),
+            does_not_raise(),
+            id="Changing a field type from a list to a string with same value should not fail.",
+        ),
+        pytest.param(
+            ConnectorSpecification(
+                connectionSpecification={
+                    "type": "object",
+                    "properties": {
+                        "my_int": {"type": ["str"]},
+                    },
+                }
+            ),
+            ConnectorSpecification(
+                connectionSpecification={
+                    "type": "object",
+                    "properties": {
+                        "my_int": {"type": ["int"]},
+                    },
+                }
+            ),
+            pytest.raises(AssertionError),
+            id="Changing a field type in list should fail.",
+        ),
+        pytest.param(
+            ConnectorSpecification(
+                connectionSpecification={
+                    "type": "object",
+                    "properties": {
+                        "my_int": {"type": ["str"]},
+                    },
+                }
+            ),
+            ConnectorSpecification(
+                connectionSpecification={
+                    "type": "object",
+                    "properties": {
+                        "my_int": {"type": ["str", "int"]},
+                    },
+                }
+            ),
+            pytest.raises(AssertionError),
+            id="Adding a field type in list should fail.",
```

Comment:
Why is this type expansion considered a breaking change here? I think if we're intentionally allowing multiple types in the spec (though maybe it's not the best practice) it's not necessarily breaking.

Reply:
Fixed in 27a2f65277db2c3943d4477d5552e32806c0b509 
I commented out the existing assertion to keep it in case we want to enable this in the future.